js/src/hooks/useNavigateAwayPromptEffect.js

```diff
@@ -18,32 +18,38 @@ export default function useNavigateAwayPromptEffect(
 	blockedLocation = () => true
 ) {
 	useEffect( () => {
-		// Bind beforeunload event for non `woocommerce/navigation` links and reloads.
-		// history#v5 does bind `beforeunload` automatically, with v4 we need to do it ourselves.
-		const eventListener = ( e ) => {
-			// If you prevent default behavior in Mozilla Firefox prompt will always be shown.
-			e.preventDefault();
-			// Chrome requires returnValue to be set.
-			e.returnValue = message;
-		};
+		// Block woocommerce/navigation in order to show a confirmation prompt in case shouldBlock is true
+		const unblock = getHistory().block( ( transition ) => {
+			let shouldUnblock = true;
 
-		if ( shouldBlock ) {
-			window.addEventListener( 'beforeunload', eventListener );
-		}
```

Comment:
:question: 
Does your change still support non `woocommerce/navigation` links and reloads with v4?

Reply:
It works with WC 6.6 that as far as I know it uses v4.

Reply:
Seems reloads and tab closes does not work https://github.com/woocommerce/google-listings-and-ads/pull/1580#issuecomment-1176453957

Reply:
Fixed here (Tested in WC 6.5, WC 6.6 & WC 6.7)   https://github.com/woocommerce/google-listings-and-ads/pull/1580/commits/f5c84ca525c25b2622338e52747c0c3962fd9c95

Thanks for the catch btw